### PR TITLE
Add gbenchmark for cudf::strings::translate function

### PR DESCRIPTION
**Pull request by davidwendt**
Reference #5698
This creates a gbenchmark for the `cudf::strings::translate()` API. The benchmarks measures various sized rows as well as strings lengths and translate table sizes.
This PR also includes changes to `translate.cu` implementation cleaning up the code and using the more efficient make_strings_children. This change improved performance for all 4 functions on average by 2-3x.
A further improvement was to sort the translation table input to more quickly lookup matches in device code. This added another 2x improvement when using longer translate tables.

### CODE REVIEW

File: cpp/benchmarks/string/translate_benchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+#include <benchmarks/common/generate_benchmark_input.hpp>
+#include <benchmarks/fixture/benchmark_fixture.hpp>
+#include <benchmarks/synchronization/synchronization.hpp>
+
+#include <cudf/strings/strings_column_view.hpp>
+#include <cudf/strings/translate.hpp>
+#include <cudf_test/column_wrapper.hpp>
+
+#include <algorithm>
+#include <random>
```
Comment (ttnghia) [inline cpp/benchmarks/string/translate_benchmark.cpp:29]: Is this header necessary?

File: cpp/benchmarks/string/translate_benchmark.cpp [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  cudf::size_type const max_str_length{static_cast<cudf::size_type>(state.range(1))};
+  data_profile table_profile;
+  table_profile.set_distribution_params(
+    cudf::type_id::STRING, distribution_id::NORMAL, 0, max_str_length);
+  auto const table =
+    create_random_table({cudf::type_id::STRING}, 1, row_count{n_rows}, table_profile);
+  cudf::strings_column_view input(table->view().column(0));
+
+  std::vector<entry_type> entries(entry_count);
+  std::transform(thrust::counting_iterator<int>(0),
```
Comment (ttnghia) [inline cpp/benchmarks/string/translate_benchmark.cpp:50]: What is the different between `std::transform` and `thrust::transform`? As I know, the C++ header <algorithm> is very heavy.

**Comment by davidwendt** [comment]
rerun tests

File: cpp/benchmarks/string/translate_benchmark.cpp (hunk 2 above)
Comment (davidwendt) [reply cpp/benchmarks/string/translate_benchmark.cpp:50]: I would rather use `std` algorithms in `.cpp` files when possible.

**Review by ttnghia — APPROVED** [review]

**Review by harrism — APPROVED** [review]

**Comment by harrism** [comment]
@gpucibot merge

**Pull request merged**